plugins/XmlMaterialProfile/XmlMaterialProfile.py

```diff
@@ -648,6 +645,163 @@ def deserialize(self, serialized):
         for container_to_add in containers_to_add:
             ContainerRegistry.getInstance().addContainer(container_to_add)
 
+    @classmethod
+    def deserializeMetadata(cls, serialized: str, container_id: str) -> List[Dict[str, Any]]:
```

Comment:
Is it possible for deserialize to either use the result of deserializeMetadata or share some parsing code?

Reply:
I've tried, but then `deserialize` needs to basically go through each of these XML tags again and basically still use the same code structure, and then also match which metadata from `deserializeMetadata` to use for which subcontainer in `deserialize`. This made for less code.